### PR TITLE
Dockerfile updates, Docker image push and support for the latest CLI

### PR DESCRIPTION
**Pull request by anzoman**
This PR will update the action with the following things:

- Dockerfile updates - minimize and optimize, use official CLI Docker image [registry.gitlab.com/xlab-steampunk/spotter-cli](https://gitlab.com/xlab-steampunk/spotter-cli/container_registry/3453459),
- Functionality updates - use the latest CLI features - uploading values and metadata, selecting target project,
- CI/CD updates - add workflow for building [ghcr.io/xlab-steampunk/spotter-action](https://github.com/xlab-steampunk/spotter-action/pkgs/container/spotter-action) Docker image for this GitHub Action,
- Use the aforementioned `spotter-action` Docker image  instead of Dockerfile and thereby speed up the action.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,13 +36,15 @@ steps:
...
+| `cli_version`     | no       | /       | The version of the [steampunk-spotter]. If not specified, the latest version is used.                                                                                              |
+| `paths`           | no       | .       | List of paths to Ansible content files to be scanned. If not specified, the whole repository is scanned.                                                                           |
+| `project_id`      | no       | /       | ID of an existing target project in the app, where the scan result will be stored. If not specified, the first project of the user's first organization (in the app) will be used. |
+| `upload_values`   | no       | false   | Parses and uploads values from Ansible task parameters to the backend.                                                                                                             |                                
+| `upload_metadata` | no       | false   | Uploads metadata (i.e., file names, line and column numbers) to the backend.                                                                                                       |
```
Comment (sstanovnik) [inline README.md:45]: Shouldn't this default to true? Ince the main usecase is scanning public repos?

File: entrypoint.sh [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-parse_values="$3"
-display_level="$4"
-ansible_version="$5"
+project_id="$3"
+upload_values="$4"
+upload_metadata="$5"
+display_level="$6"
+ansible_version="$7"
 
 installDifferentSpotterCLIVersionIfNeeded() {
```
Comment (sstanovnik) [inline entrypoint.sh:12]: This doesn't need to be exist since we're using a Docker image, right?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-20.04
+    if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags')
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
```
Comment (sstanovnik) [inline .github/workflows/docker.yml:14]: I might be missing something important, but why do you need qemu?

**Review by sstanovnik — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 2 above)
Comment (anzoman) [reply entrypoint.sh:12]: True yes, we do use a Docker image for the CLI, but changing the CLI version cannot be applied by changing the CLI Docker image version dynamically, so if user wants different CLI we need to do it here.

But all in all maybe we don't even need this and can always use just the latest CLI version, @gberginc what do you think?

File: .github/workflows/docker.yml (hunk 3 above)
Comment (anzoman) [reply .github/workflows/docker.yml]: Ah yes, I was in need of this for some other project, so here we probably don't need it and I removed it.

File: entrypoint.sh (hunk 2 above)
Comment (anzoman) [reply entrypoint.sh]: Updates to use only the latest version.

File: README.md (hunk 1 above)
Comment (matejart) [reply README.md:44]: To me `false` makes sense as the default choice, because the sending of any private data should be a deliberate opt in.

**Review by sstanovnik — APPROVED** [review]

**Comment by sstanovnik** [comment]
Looks updated to me!

**Pull request merged**